src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java

```diff
@@ -1798,4 +1811,50 @@ public static boolean isWithinRect(int x, int y, int left, int top, int width, i
 		return left <= x && x <= left + width &&
 			top <= y && y <= top + height;
 	}
+
+	public static int getNumberOfStars(ItemStack stack) {
+		if (stack != null && stack.hasTagCompound()) {
+			NBTTagCompound tag = stack.getTagCompound();
+
+			if (tag.hasKey("ExtraAttributes", 10)) {
+				NBTTagCompound ea = tag.getCompoundTag("ExtraAttributes");
+				if (ea.hasKey("upgrade_level", 99)) {
+					return ea.getInteger("upgrade_level");
+				} else if (ea.hasKey("dungeon_item_level")) {
+					return ea.getInteger("dungeon_item_level");
+				}
+			}
+		}
+		return 0;
+	}
+
+	public static String getStarsString(int stars, String displayName) {
+		//ignore master stars
+		if (displayName.contains("§c")) {
+			return null;
+		}
+
+		String colorCode = null;
+		int amount = 0;
+		if (stars > 5 && stars < 11) {
+			colorCode = "§d";
+			amount = stars - 5;
+			stars = 5;
+		}
+		if (stars > 10) {
+			colorCode = "§b";
+			amount = stars - 10;
+			stars = 5;
+		}
+
+		StringBuilder stringBuilder = new StringBuilder();
+		for (int i = 0; i < stars; i++) {
+			if (i < amount) {
+				stringBuilder.append(colorCode).append('\u272A');
+			} else {
+				stringBuilder.append(EnumChatFormatting.GOLD).append(EnumChatFormatting.BOLD).append('\u272A');
```

Comment:
This is the wrong behaviour for items with 11 to 14 stars, coloring the stars to the right gold instead of what they should be. Also i dont think the stars are bold, however i could be wrong with that one.